server/circleci.go

```diff
@@ -49,15 +51,16 @@ func (s *Server) triggerCircleCiIfNeeded(ctx context.Context, pr *model.PullRequ
 		return
 	}
 
-	for _, prFile := range prFiles {
-		for _, blockListPath := range s.Config.BlacklistPaths {
-			if prFile.GetFilename() == blockListPath {
-				mlog.Error("File is on the blocklist and will not retrigger circleci to give the contexts", mlog.String("repo", pr.RepoName), mlog.Int("pr", pr.Number), mlog.String("Fullname", pr.FullName))
-				msg := fmt.Sprintf("The file `%s` is in the blocklist and should not be modified from external contributors, please if you are part of the Mattermost Org submit this PR in the upstream.\n /cc @mattermost/core-security @mattermost/core-build-engineers", prFile.GetFilename())
-				s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, msg)
-				return
-			}
+	blockMessage, err := s.validateBlockPaths(prFiles)
+	if err != nil {
+		if blockMessage != "" {
```

Comment:
I think we can improve the error handling a bit further by having a separate error type for blockList error and not returning any `blockMessage` at all. What we are trying to do here is check whether there is an error or not, then differentiate whether it's a blocklist error or not by inspecting the `blockMessage` variable.

IMO, if we have a generic error for the filepath failure case, and a separate error type for an actual blockList error, we can handle this case more elegantly.

Reply:
did some changes after our chat. PTAL
I think that maybe still not in the way you want, but was what I understood :)